wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -487,6 +487,29 @@ public void createSnapshot(String workspaceId) throws NotFoundException, ServerE
         return machineManager.getSnapshots(sessionUser().getId(), workspaceId);
     }
 
+    /**
+     * Adds machine into workspace runtime.
+     *
+     * @param machine
+     *         machine to add to specified runtime
+     * @throws ConflictException
+     *         if given machine already exists in specified runtime
+     *         or if machine is dev-machine
+     * @throws NotFoundException
+     *         if machine's workspace hasn't runtime
+     */
+    public void injectMachine(MachineImpl machine) throws ConflictException, NotFoundException, ServerException {
```

Comment:
I remember that i suggested you to consider either `addMachine` or `injectMachine` now i see that, `addMachine` is actually better name, just because we will need another method for removing machine from the running workspace, so there is no good opposite word for `inject`, but there is a good opposite word for `add`. Can you please rename it to the `addMachine`.